test/end-to-end.js

```diff
@@ -0,0 +1,307 @@
+         logReceipt(o.receipt, "Trustee.reclaimTokens")
+      })
+
+      it("Transfer future sale tokens to lock box ", async () => {
+         const totalPresaleBonus = await sale.totalPresaleBonus.call()
+
+         const futureTokens = TOKENS_FUTURE.sub(totalPresaleBonus)
+
+         await token.transfer(lockBox.address, futureTokens, { from: deployKey })
+      })
+
+      it("Transfer remaining tokens to accelerator wallet(s)", async () => {
+         const tokensLeft = await token.balanceOf.call(deployKey)
+
+         await token.transfer(accelerator, tokensLeft, { from: deployKey })
+      })
+
+      it("Transfer ownership of trustee contract", async () => {
+         const o1 = await trustee.initiateOwnershipTransfer(trusteeOwner, { from: deployKey })
+         const o2 = await trustee.completeOwnershipTransfer({ from: trusteeOwner })
+         logReceipt(o1.receipt, "Owned.initiateOwnershipTransfer")
+         logReceipt(o2.receipt, "Owned.completeOwnershipTransfer")
+      })
+
+      it("Transfer ownership of sale contract", async () => {
+         await sale.initiateOwnershipTransfer(saleOwner, { from: deployKey })
+         await sale.completeOwnershipTransfer({ from: saleOwner })
+      })
+
+      it("Transfer ownership of lock box contract", async () => {
+         await lockBox.initiateOwnershipTransfer(lockBoxOwner, { from: deployKey })
+         await lockBox.completeOwnershipTransfer({ from: lockBoxOwner })
+      })
+   })
+
+
+   describe('Whitelisting phase', async () => {
+
+      it("Add person 1 to phase 1 whitelist", async () => {
+         const o = await sale.updateWhitelist(whitelisted1, 1, { from: whitelistService })
+         logReceipt(o.receipt, "TokenSale.updateWhitelist")
+      })
+
+      it("Add person 2 to phase 2 whitelist", async () => {
+         await sale.updateWhitelist(whitelisted2, 2, { from: whitelistService })
+      })
+   })
+
+
+   describe('Before sale phase', async () => {
+
+      it("Set price for tokens", async () => {
+         const price = new BigNumber(300000).div(0.1667)
+
+         const o = await sale.setTokensPerKEther(price, { from: admin })
+         logReceipt(o.receipt, "TokenSale.setTokensPerKEther")
+      })
+   })
+
+
+   describe('During sale phase', async () => {
+
+      it("whitelisted 1 buys during phase 1", async () => {
+         await sale.changeTime(PHASE1_START_TIME, { from: saleOwner })
+
+         const o = web3.eth.sendTransaction({ from: whitelisted1, to: sale.address, value: web3.toWei(1, 'ether') })
+         await logTransaction(o, "sendTransaction (buy tokens)")
+      })
+
+      it("whitelisted 2 buys during phase 2", async () => {
+         await sale.changeTime(PHASE2_START_TIME, { from: saleOwner })
+
+         web3.eth.sendTransaction({ from: whitelisted2, to: sale.address, value: web3.toWei(1, 'ether') })
+      })
+
+   })
+
+
+   describe('Sale finalization phase', async () => {
+
+      it("finalize the sale", async () => {
```

Comment:
Question where I will defer to your opinion on best practice: our ideal scenario is that `finalize` is called automatically and there are no tokens to `reclaim`. Does it make sense for the end-to-end to model what we intend will happen?

So: (a) set the phase1 max as high as possible (so, contribution max?) in the 'Before sale phase', thus including a function we plan to call; (b) have whitelisted1/2 call 6 times each with the max contribution (which would then include the refund functionality which is also likely to be the case) in the 'Sale finalization phase'; and (c) also in the 'Sale finalization phase', call the `reclaimTokens` function (because transfers of 0 are now allowed and maybe it's a good redundancy practice?)